### PR TITLE
Update PoolingRegion to work with Network API (specifically in NLP research)

### PR DESCRIPTION
**Pull request by chetan51**

### CODE REVIEW

**Comment by chetan51** [comment]
@BoltzmannBrain @subutai Please review that these changes make sense.

File: htmresearch/regions/PoolingRegion.py [hunk 1]
```diff
@@ -243,7 +249,14 @@ def compute(self, inputs, outputs):
     The guts of the compute are contained in the self._poolerClass compute() call
     """
     activeCells = inputs["activeCells"]
-    predictedActiveCells = inputs["predictedActiveCells"]
+    predictedActiveCells = inputs["predictedActiveCells"] if (
```
Comment (subutai) [inline htmresearch/regions/PoolingRegion.py]: If this is a required input, should we throw an exception if not connected? If so, the exception can be thrown in initialize, which is always called once before the first all to compute().

File: htmresearch/regions/PoolingRegion.py [hunk 2]
```diff
@@ -243,7 +249,14 @@ def compute(self, inputs, outputs):
...
+      "predictedActiveCells" in inputs) else numpy.zeros(self._inputWidth)
+
+    resetSignal = False
+    if 'resetIn' in inputs:
+      assert len(inputs['resetIn']) == 1
```
Comment (subutai) [inline htmresearch/regions/PoolingRegion.py]: This check can be in initialize() and should throw an exception not an assert.

**Comment by subutai** [comment]
:+1:

File: htmresearch/regions/PoolingRegion.py (hunk 1 above)
Comment (chetan51) [inline htmresearch/regions/PoolingRegion.py]: This is not a required input. In fact in the NLP experiment that uses pooling, it's connected to a sensor directly, which wouldn't produce predicted active cells.

**Pull request merged**